### PR TITLE
Refactor roomMember Providers, user info drawer

### LINKED ISSUE #1049: mention user in chat, doesnt always seem to find freshly changed/joined user
https://github.com/acterglobal/a3/assets/77483554/22f3ca93-ffec-402b-8025-f5c486805804

### PR DESCRIPTION
**Pull request by gnunicorn**
Massively refactors the space/convo membership concept to:
1. only provider userIds for active users
2. reuse one single room-based membership provider for all profile data
3. drive-along fix for the way mentions for the user-listing are generated and what is searched for (includes the userId regardless of displayname set), fixes #1049

### CODE REVIEW

File: app/lib/common/dialogs/member_info_drawer.dart [hunk 1]
```diff
@@ -13,18 +14,61 @@ import 'package:flutter/material.dart';
...
-class _MemberInfoDrawer extends ConsumerWidget {
-  final ProfileData memberProfile;
-  final String memberId;
+class _MessageUser extends ConsumerWidget {
   final Member member;
-  final String roomId;
-  const _MemberInfoDrawer({
-    super.key,
-    required this.memberProfile,
+  const _MessageUser({required this.member});
```
Comment (gnunicorn) [inline app/lib/common/dialogs/member_info_drawer.dart:21]: just moved that into a separate widget. all else is a copy-paste.

File: app/lib/common/models/types.dart [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import 'package:acter/common/models/profile_data.dart';
```
Comment (gnunicorn) [inline app/lib/common/models/types.dart:1]: this is a pure move from the utils into a more meaningful file.

File: app/lib/common/dialogs/member_info_drawer.dart [hunk 3]
```diff
@@ -174,12 +178,108 @@ class _MemberInfoDrawer extends ConsumerWidget {
   }
 }
 
+class _MemberInfoSkeleton extends StatelessWidget {
```
Comment (kumarpalsinh25) [inline app/lib/common/dialogs/member_info_drawer.dart:181]: Good to have it on separate skeleton file and folder (If have multiple skeletons).

File: app/lib/common/dialogs/member_info_drawer.dart [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import 'package:acter/common/dialogs/block_user_dialog.dart';
```
Comment (kumarpalsinh25) [inline app/lib/common/dialogs/member_info_drawer.dart:1]: I see there there are multiple class related to memberInfo features in this single file (`app/lib/common/dialogs/member_info_drawer.dart)`. I would suggest to create `member_info` folder and put different class in different files for better code readability.

File: app/lib/common/widgets/member_list_entry.dart [hunk 5]
```diff
@@ -176,17 +177,84 @@ class _ChangePowerLevelState extends State<ChangePowerLevel> {
...
-  final Convo? convo;
+  final String memberId;
+  final String roomId;
   final Member? myMembership;
 
   const MemberListEntry({
     super.key,
+    required this.memberId,
+    required this.roomId,
+    this.myMembership,
```
Comment (kumarpalsinh25) [inline app/lib/common/widgets/member_list_entry.dart:225]: Why to have `myMembership` in constructer even thought we can use `final myMembership = ref.watch(roomMembershipProvider(roomId));` in this class?

Comment (kumarpalsinh25) [reply app/lib/common/widgets/member_list_entry.dart:225]: Better to have minimus argument to standalone component to make it independent of data dependency.

**Review by kumarpalsinh25 — APPROVED** [review]
Very nice! Much simplified code for `roomMemebrProviders`.

Add minor suggestions but you can do that on separate PR if needed. :)

File: app/lib/common/dialogs/member_info_drawer.dart (hunk 4 above)
Comment (gnunicorn) [reply app/lib/common/dialogs/member_info_drawer.dart:1]: They are all _intentionally_ internal (prefixed with a `_`). If I put them into several files or folders I would have to make them public and but I don't want any API public other than the `showMemberInfoDrawer()` fn. That's why I will leave it as is.
`

**Pull request merged**